packages/astro/components/Outlet.astro

```diff
@@ -0,0 +1,14 @@
+---
+interface Props {
+	id: string;
+}
+
+const { id } = Astro.props;
+
+// @ts-ignore untyped internals
+$$result.outlets.set(id, () => Astro.slots.render('default'));
+---
+
+<Fragment set:html={`<!--astro:outlet ${id}-->`} /><slot /><Fragment
+	set:html={`<!--/astro:outlet ${id}-->`}
+/>
```

Comment:
`Outlet` itself is just a wrapper that marks the boundaries of the component with comments